mllib/src/main/scala/org/apache/spark/ml/util/Identifiable.scala

```diff
@@ -19,15 +19,24 @@ package org.apache.spark.ml.util
 
 import java.util.UUID
 
+
 /**
- * Object with a unique id.
+ * Trait for an object with an immutable unique ID that identifies itself and its derivatives.
  */
-private[ml] trait Identifiable extends Serializable {
+trait Identifiable {
+
+  /**
+   * An immutable unique ID for the object and its derivatives.
```

Comment:
How user-facing is this?  Should we provide more info/promises about how it is used, or should we wait until that is clearer to us?